### PR TITLE
Animations in the lab and code cleanup

### PR DESCRIPTION
**Pull request by The-E**
This allows animations to be triggered in the lab. It also changes the TRIGGER_TYPE_* defines to an enum class and performs a bit of code cleanup in the lab.
Furthermore, this introduces a basic grouping mechanism for weapons when changing weapons in the actions menu.

### CODE REVIEW

File: code/model/modelanim.cpp [hunk 1]
```diff
@@ -22,70 +22,67 @@ SCP_vector<triggered_rotation> Triggered_rotations;
...
 	} else if ( !strnicmp(p, "door", 4) || !strnicmp(p, "\"door\"", 6) ) {
-		mprintf(("The \"door\" animation type name is deprecated.  Specify \"%s\" instead.\n", Animation_type_names[TRIGGER_TYPE_DOCK_BAY_DOOR]));
-		return TRIGGER_TYPE_DOCK_BAY_DOOR;
+		mprintf(("The \"door\" animation type name is deprecated.  Specify \"fighterbay\" instead.\n"));
+		return AnimationTriggerType::DockBayDoor;
 	} else if ( !strnicmp(p, "turret firing", 13) || !strnicmp(p, "\"turret firing\"", 15) ) {
-		mprintf(("The \"turret firing\" animation type name is deprecated.  Specify \"%s\" instead.\n", Animation_type_names[TRIGGER_TYPE_TURRET_FIRING]));
-		return TRIGGER_TYPE_TURRET_FIRING;
+		mprintf(("The \"turret firing\" animation type name is deprecated.  Specify \"turret-firing\" instead.\n"));
+		return AnimationTriggerType::TurretFiring;
```
Comment (Goober5000) [inline code/model/modelanim.cpp:69]: Why remove the `Animation_type_names` argument?  That removes any chance of misspelling (not that you misspelled anything), and automatically updates the warning if for some reason the name changes in the future.

File: code/model/modelanim.cpp [hunk 2]
```diff
@@ -98,7 +95,7 @@ void triggered_rotation::start(queued_animation *q)
 {
 	instance = q->instance;
 
-	nprintf(("ModelAnim", "Starting animation type %i at %i ...\n", q->type, timestamp()));
+	nprintf(("ModelAnim", "Starting animation type %i at %i ...\n", static_cast<int>(q->type), timestamp()));
```
Comment (Goober5000) [inline code/model/modelanim.cpp:98]: You could use `Animation_type_names` here

File: code/lab/lab.cpp [hunk 3]
```diff
@@ -2093,32 +2097,41 @@ void lab_weapon_fire_window_close(GUIObject* /*caller*/) {
...
+		if (obj->type == OBJ_SHIP) {
+			obj->flags.remove(Object::Object_Flags::Player_ship);
+			ship_self_destruct(obj);
+			Lab_viewer_flags |= LAB_FLAG_SHIP_EXPLODING;
+		}
 	}
 }
 
+// Returns true if it is safe to access the Ships array
+bool actions_window_is_safe() {
```
Comment (Goober5000) [inline code/lab/lab.cpp:2117]: Call the function `actions_window_safe_for_ships` or something for future proofing.  There might be non-ship actions in the future.

**Review by Goober5000 — CHANGES_REQUESTED** [review]
A couple comments

**Comment by The-E** [comment]
Comments have been addressed

**Review by Goober5000 — APPROVED** [review]

**Pull request merged**